google-beta/resource_compute_instance_group_manager.go

```diff
@@ -373,38 +351,45 @@ func flattenFixedOrPercent(fixedOrPercent *computeBeta.FixedOrPercent) []map[str
 func getManager(d *schema.ResourceData, meta interface{}) (*computeBeta.InstanceGroupManager, error) {
 	config := meta.(*Config)
 
-	project, err := getProject(d, config)
+	id, err := parseInstanceGroupManagerId(d.Id())
 	if err != nil {
 		return nil, err
 	}
-
-	region, err := getRegion(d, config)
-	if err != nil {
-		return nil, err
+	if id.Project == "" {
+		project, err := getProject(d, config)
+		if err != nil {
+			return nil, err
+		}
+		id.Project = project
+	}
+	if id.Zone == "" {
+		id.Zone, _ = getZone(d, config)
 	}
 
 	getInstanceGroupManager := func(zone string) (interface{}, error) {
-		return config.clientComputeBeta.InstanceGroupManagers.Get(project, zone, d.Id()).Do()
+		return config.clientComputeBeta.InstanceGroupManagers.Get(id.Project, zone, id.Name).Do()
 	}
 
 	var manager *computeBeta.InstanceGroupManager
-	var e error
-	if zone, _ := getZone(d, config); zone != "" {
-		manager, e = config.clientComputeBeta.InstanceGroupManagers.Get(project, zone, d.Id()).Do()
-
-		if e != nil {
-			return nil, handleNotFoundError(e, d, fmt.Sprintf("Instance Group Manager %q", d.Get("name").(string)))
+	if id.Zone != "" {
+		manager, err = config.clientComputeBeta.InstanceGroupManagers.Get(id.Project, id.Zone, id.Name).Do()
+		if err != nil {
+			return nil, handleNotFoundError(err, d, fmt.Sprintf("Instance Group Manager %q", id.Name))
 		}
 	} else {
 		// If the resource was imported, the only info we have is the ID. Try to find the resource
 		// by searching in the region of the project.
-		var resource interface{}
-		resource, e = getZonalBetaResourceFromRegion(getInstanceGroupManager, region, config.clientComputeBeta, project)
-		if e != nil {
-			return nil, e
+		region, err := getRegion(d, config)
```

Comment:
oh that's a bummer. you could also do a state migration, though i'm not sure that would actually be simpler.

Reply:
Yeah, I just recall that state migrations were a pain to test. I figure we could start with this, get things in the right shape, and worst case scenario, we could do a state migration and simplify later, when we had some more time to test.

Reply:
Wait won't any igm in state already have `zone` set? If we read `zone` `name` and `project` from state instead of parsing the `id`, existing igms in state will work & the only breaking change will be that scripts that import by region will break.

Reply:
It will, but there's no guarantee that things like Delete or even read have access to the config still. In some weird scenarios, I know Delete doesn't. Not 100% sure Read doesn't.

Reply:
why would they need access to the config (vs state)?

Reply:
Ohhhh, fair.

Following this release, let's look into just doing the state migration (if necessary) and just deleting this awk code.